cartoframes/data/observatory/catalog/dataset.py

```diff
@@ -379,6 +382,8 @@ def download(self, credentials=None):
         :raises CartoException: If you have not a valid license for the dataset being downloaded.
         :raises ValueError: If the credentials argument is not valud.
         """
+        if not self._is_subscribed(credentials):
+            raise CartoException('You are not subscribed to this Dataset yet. Please, use the subscribe method first.')
```

Comment:
Why a `CartoException` (from carto-python) instead of an `Exception`

Reply:
Because we are using it as a default error exception here too when the error comes from something related to CARTO

Reply:
OK. We will review all the exceptions in the lib the next iteration, so it's OK now